bindinfo/bind.go

```diff
@@ -15,155 +15,66 @@ package bindinfo
+}
 
-	if originalNode.Fields != nil {
-		origFields := originalNode.Fields.Fields
-		hintFields := hintedNode.Fields.Fields
-		for idx := range origFields {
-			origFields[idx].Expr = exprBind(origFields[idx].Expr, hintFields[idx].Expr)
-		}
+func (hc *hintCollector) Enter(in ast.Node) (ast.Node, bool) {
+	switch v := in.(type) {
+	case *ast.SelectStmt:
+		hc.tableHints[hc.tableCounter] = v.TableHints
+		hc.tableCounter++
+	case *ast.TableName:
+		hc.indexHints[hc.indexCounter] = v.IndexHints
+		hc.indexCounter++
 	}
-	return originalNode
+	return in, false
 }
 
-func orderByBind(originalNode, hintedNode *ast.OrderByClause) *ast.OrderByClause {
-	for idx := 0; idx < len(originalNode.Items); idx++ {
-		originalNode.Items[idx].Expr = exprBind(originalNode.Items[idx].Expr, hintedNode.Items[idx].Expr)
-	}
-	return originalNode
+func (hc *hintCollector) Leave(in ast.Node) (ast.Node, bool) {
+	return in, true
 }
 
-func exprBind(originalNode, hintedNode ast.ExprNode) ast.ExprNode {
-	switch v := originalNode.(type) {
-	case *ast.SubqueryExpr:
-		if v.Query != nil {
-			v.Query = resultSetNodeBind(v.Query, hintedNode.(*ast.SubqueryExpr).Query)
-		}
-	case *ast.ExistsSubqueryExpr:
-		if v.Sel != nil {
-			v.Sel.(*ast.SubqueryExpr).Query = resultSetNodeBind(v.Sel.(*ast.SubqueryExpr).Query, hintedNode.(*ast.ExistsSubqueryExpr).Sel.(*ast.SubqueryExpr).Query)
-		}
-	case *ast.PatternInExpr:
-		if v.Sel != nil {
-			v.Sel.(*ast.SubqueryExpr).Query = resultSetNodeBind(v.Sel.(*ast.SubqueryExpr).Query, hintedNode.(*ast.PatternInExpr).Sel.(*ast.SubqueryExpr).Query)
-		}
-	case *ast.BinaryOperationExpr:
-		if v.L != nil {
-			v.L = exprBind(v.L, hintedNode.(*ast.BinaryOperationExpr).L)
-		}
-		if v.R != nil {
-			v.R = exprBind(v.R, hintedNode.(*ast.BinaryOperationExpr).R)
-		}
-	case *ast.IsNullExpr:
-		if v.Expr != nil {
-			v.Expr = exprBind(v.Expr, hintedNode.(*ast.IsNullExpr).Expr)
-		}
-	case *ast.IsTruthExpr:
-		if v.Expr != nil {
-			v.Expr = exprBind(v.Expr, hintedNode.(*ast.IsTruthExpr).Expr)
-		}
-	case *ast.PatternLikeExpr:
-		if v.Pattern != nil {
-			v.Pattern = exprBind(v.Pattern, hintedNode.(*ast.PatternLikeExpr).Pattern)
-		}
-	case *ast.CompareSubqueryExpr:
-		if v.L != nil {
-			v.L = exprBind(v.L, hintedNode.(*ast.CompareSubqueryExpr).L)
-		}
-		if v.R != nil {
-			v.R = exprBind(v.R, hintedNode.(*ast.CompareSubqueryExpr).R)
-		}
-	case *ast.BetweenExpr:
-		if v.Left != nil {
-			v.Left = exprBind(v.Left, hintedNode.(*ast.BetweenExpr).Left)
-		}
-		if v.Right != nil {
-			v.Right = exprBind(v.Right, hintedNode.(*ast.BetweenExpr).Right)
-		}
-	case *ast.UnaryOperationExpr:
-		if v.V != nil {
-			v.V = exprBind(v.V, hintedNode.(*ast.UnaryOperationExpr).V)
-		}
-	case *ast.CaseExpr:
-		if v.Value != nil {
-			v.Value = exprBind(v.Value, hintedNode.(*ast.CaseExpr).Value)
-		}
-		if v.ElseClause != nil {
-			v.ElseClause = exprBind(v.ElseClause, hintedNode.(*ast.CaseExpr).ElseClause)
-		}
-	}
-	return originalNode
+type hintBinder struct {
```

Comment:
can we merge `hintBinder` and `hintCollector` together? To avoid duplicated code.